### PR TITLE
Feature/sft fix firmware revision handling

### PR DESCRIPTION
**Pull request by atigyi**
[ Description ]
Fixing a rollback firmware revision issue. In case the new firmware failed thus rollback happened the old firmware did use the new firmware's revision. This change solves the issue, now the rollbacked firmware does not use the new firmware's revision but its old one.

[ JIRA ]
design+impl for Firmware's revision handling
XCL-3060

[ Reviewer ]
@DellaBitta

[ QA ]
Posix FWU process intentionally made to fail and checked whether rollback used the old revision.

[ Release Notes ]
n/a

### CODE REVIEW

File: src/bsp/platform/posix/xi_bsp_fwu_posix.c [hunk 1]
```diff
@@ -45,21 +96,42 @@ void xi_bsp_fwu_on_package_download_finished( const char* const firmware_resourc
...
+    }
+
+    /* - rename file to firmware_resource_name ## _DATETIMESTAMP */
     const char* update_fw_name = _generate_update_filename( firmware_resource_name );
     rename( firmware_resource_name, update_fw_name );
 
 #if 0
-    // - start new firmware
+    {
+        /* - save the new firmware progname to identify new firmware first run
```
Comment (DellaBitta) [inline src/bsp/platform/posix/xi_bsp_fwu_posix.c]: program.

File: src/bsp/platform/posix/xi_bsp_fwu_posix.c [hunk 2]
```diff
@@ -45,21 +96,42 @@ void xi_bsp_fwu_on_package_download_finished( const char* const firmware_resourc
...
-                             "-a 4d3c7986-8d53-4cf8-903e-7fd30ff63be1 "
-                             "-u fc4ba77e-0ba0-4ef2-bb46-d30bc59546ab "
-                             "-P JMnzPt3zDH2ZY2Q/ZQRgzublMGF6VfAFv7vFxFf5tLY= "
-                             "-p xi/blue/v1/4d3c7986-8d53-4cf8-903e-7fd30ff63be1/d/"
-                             "b03e863d-2478-48cf-976e-fa2d5ddf4ec8/Temperature",
+                             "-a <Xively Account ID> "
+                             "-u <Xively Device ID> "
+                             "-P <Xively Device Password> "
+                             "-p xi/blue/v1/<Xively Account ID>/d/"
+                             "<Xively Device ID>/Temperature",
```
Comment (DellaBitta) [inline src/bsp/platform/posix/xi_bsp_fwu_posix.c]: Hold on, writing these down...

Comment (DellaBitta) [reply src/bsp/platform/posix/xi_bsp_fwu_posix.c]: oh so we DO still need the account id?

Comment (atigyi) [reply src/bsp/platform/posix/xi_bsp_fwu_posix.c]: seems so, yes. Maybe control topic does not need accountid, but apparently "other" topic names include it.

Comment (atigyi) [reply src/bsp/platform/posix/xi_bsp_fwu_posix.c]: but still the xi_initialize needs the username despite it is passed in xi_connect too later.

File: src/bsp/platform/posix/xi_bsp_fwu_posix.c (hunk 1 above)
Comment (atigyi) [reply src/bsp/platform/posix/xi_bsp_fwu_posix.c]: changed to "executable name"

**Pull request merged**